tests/job/test_genericJob.py

```diff
@@ -502,5 +504,56 @@ def test_return_codes(self):
             pass
         self.assertTrue(j.status.aborted, "Job did not abort even though return code is 2!")
 
+    def test_job_executor_set(self):
+        j = self.project.create_job(ReturnCodeJob, "job_with_executor_set")
+        j.server.executor = ProcessPoolExecutor()
+        self.assertTrue(j.server.run_mode.executor)
+        j.server.run_mode.modal = True
+        self.assertFalse(j.server.run_mode.executor)
+
+    def test_job_executor_run(self):
+        j = self.project.create_job(ReturnCodeJob, "job_with_executor_run")
+        j.input["accepted_codes"] = [1]
+        j.server.executor = ProcessPoolExecutor()
+        self.assertTrue(j.server.run_mode.executor)
+        j.run()
+        j.server.future.result()
+        self.assertTrue(j.server.future.done())
+        self.assertTrue(j.status.finished)
+
+    def test_job_executor_cancel(self):
+        j = self.project.create_job(ReturnCodeJob, "job_with_executor_cancel")
+        j.input["accepted_codes"] = [1]
+        exe = ProcessPoolExecutor()
+        j.server.executor = exe
+        self.assertTrue(j.server.run_mode.executor)
+        exe.submit(sleep, 1)  # This part is a bit hacky, but it basically simulates other jobs on the same executor
+        j.run()
+        j.server.future.cancel()
+        j.refresh_job_status()
+        self.assertTrue(j.status.aborted)
+
+    def test_job_executor_wait(self):
+        j = self.project.create_job(ReturnCodeJob, "job_with_executor_wait")
+        j.input["accepted_codes"] = [1]
+        j.server.executor = ProcessPoolExecutor()
+        self.assertTrue(j.server.run_mode.executor)
+        j.run()
+        self.project.wait_for_job(job=j)
+        self.assertTrue(j.server.future.done())
+        self.assertTrue(j.status.finished)
+
+    def test_job_executor_copy(self):
+        j1 = self.project.create_job(ReturnCodeJob, "job_with_executor_copy")
+        j1.input["accepted_codes"] = [1]
+        j1.server.executor = ProcessPoolExecutor()
+        j2 = j1.copy()
+        self.assertIs(j2.server.executor, j1.server.executor)
+        self.assertTrue(j2.server.run_mode.executor)
+        j2.run()
+        j2.server.future.result()
+        self.assertTrue(j2.server.future.done())
+
```

Comment:
```suggestion
j3 = j2.copy()
self.assertIs(j3.server.future, j2.server.future)

```

Assuming this is the behaviour we want when copying finished executor jobs. I'm open to the opposite behaviour if there's a case for it that I'm not seeing.

Reply:
From my perspective the future object is only reasonable during the execution, after the execution, when the status is set to finished, then the future is no longer required and it is recommended to just check the job status.

Reply:
If it's not supposed to be looked at anymore, then we should just get rid of it

Reply:
Ok, we agree elsewhere it still needs to be around, but shouldn't copy if it's not done. That behaviour is implemented in another suggestion and tested here.